### PR TITLE
Update raindrops generator

### PR DESCRIPTION
**Pull request by bmulvihill**
Based on https://github.com/exercism/xruby/issues/412

The generator no longer relies on the description, instead using the number

The generator metadata directory also needed to be updated based on the x-common layout.

### CODE REVIEW

**Comment by Insti** [comment]
Awesome, Thanks :heart:
Someone will review it in the next few days.

File: lib/raindrop_cases.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class RaindropsCase < OpenStruct
   def name
-    'test_%s' % description.downcase.gsub(/[ -]/, '_')
+    'test_%s' % number
```
Comment (Insti) [inline lib/raindrop_cases.rb]: Where does `number` come from?

File: lib/generator.rb [hunk 2]
```diff
@@ -14,11 +14,11 @@ def initialize(name, cases)
   end
 
   def metadata_dir
-    File.expand_path(File.join('..', '..', '..', METADATA_REPOSITORY), __FILE__)
+    File.expand_path(File.join('..', '..', '..', METADATA_REPOSITORY, 'exercises', "#{ name }"), __FILE__)
```
Comment (Insti) [inline lib/generator.rb]: No need to have redundant string interpolation here, 
```
File.expand_path(File.join('..', '..', '..', METADATA_REPOSITORY, 'exercises', name), __FILE__)
```
should work just as well.

**Comment by Insti** [comment]
Ideally the generator.rb changes would be part of separate PR, since they're not raindrops specific. If you can make another PR for that it would be good.

File: lib/raindrop_cases.rb (hunk 1 above)
Comment (Insti) [inline lib/raindrop_cases.rb]: Found it, It's one of the keys in the JSON.

**Comment by Insti** [comment]
All the raindrops specific stuff looks good :+1: 

Just move the `generator.rb` changes to a new PR and fix that string interpolation and we're good to go.

(Please let me know if you need help with this.)

**Comment by bmulvihill** [comment]
@Insti Ok, I will open another PR with the generator stuff

**Comment by Insti** [comment]
The updated raindrops files should be part of this PR.

Create the generator PR and we can merge that in, then add the regenerated raindrops_test file back into this PR.

(Or you can do it locally by just not checking in the changed generator.rb file to this branch.)

Does that make sense?

**Comment by Insti** [comment]
Waiting on the generator update PR to be merged.

**Comment by Insti** [comment]
Thanks again @bmulvihill !
If you're doing Hacktoberfest you're half way to a T-shirt.

**Pull request merged**